contraint.go

```diff
@@ -62,6 +62,62 @@ type TimeConstraint interface {
 	ValidateTime(value *time.Time, scope Scope) error
 }
 
+type ConditionalConstraint struct {
```

Comment:
Кстати, нет документации

Reply:
> ConditionalConstraint is used to construct constraints using a conditional construct.

Это какая-то скороговорка? Похоже на "Ехал грека через реку..." :-) А если честно, то такая документация хуже ее полного отсутствия. Рекомендую сначала выразить мысли на русском (как бы ты хотел видеть описание, если бы изучал документацию библиотеки), а затем перевести это.

ConditionalConstraint is used for conditional validation. Use the When function to initiate a conditional check. If the condition is true, then the constraints passed through the Then function will be applied. Otherwise, the constraints passed through the Else function will apply.